src/kitovu/sync/syncing.py

```diff
@@ -56,70 +56,93 @@ def start_all(config_file: typing.Optional[pathlib.Path]) -> None:
 
 def start(connection_name: str, connection_settings: ConnectionSettings) -> None:
     """Sync files with the given plugin and username."""
-    # FIXME remove this after refactoring this function
-    # pylint: disable=too-many-locals
-
     logger.info(f'Syncing connection {connection_name}')
 
     plugin = _load_plugin(connection_settings)
-    plugin.configure(connection_settings.connection)
-    plugin.connect()
+
+    try:
+        plugin.configure(connection_settings.connection)
+        plugin.connect()
+    except utils.PluginOperationError as ex:
```

Comment:
There was a missing `return` here, which I added now.

I don't think plugins should use `UsageError` at all (or only in cases where they really want to abort everything, for some reason) - also see my changes to the documentation above.

That way, you can tell plugin authors to simply use `PluginOperationError` and kitovu will do the right thing, instead of needing to tell them to use `UsageError` in some places and `PluginOperationError` in other places.

Reply:
Alright, we can do it that way. This way it's not possible to cancel the execution in the other methods (eg. you're in a train and loose connection, it will print "No connection to server" for each remaining file). But I think we can live with that.